spec.md

```diff
@@ -454,7 +454,9 @@ Client and registry implementations SHOULD expect to be able to support manifest
 
 ###### Pushing Manifests with Subject
 
-When pushing an image or artifact manifest with the `subject` field and the [referrers API](#listing-referrers) returns a 404, the client MUST:
+When processing a request for an image or artifact manfiest with the `subject` field, a registry implementation that supports the [referrers API](#listing-referrers) MUST respond with the response header `OCI-Subject-Processed: true` to indicate to the client that the registry understands the request's `subject`.
```

Comment:
I was thinking more along the lines of:

```
OCI-Subject: sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270
```

So the registry actually has to process the subject field and extract the digest to generate the response, which guarantees that the registry _did_ actually understand the `subject` field enough to generate this response, which gives us some assurance that it did something meaningful with it.

Reply:
That's a good idea.

So just as a clarifying example, if this manifest gets pushed:

```
POST /v2/blah/blah/manifests/sha256:d54bcc13e190301b388a828178c02e3c3faa6c04a3a54bf90802c17ae892cebc
{
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "layers": [
    ...
  ],
  "subject": {
    "digest": "sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270"
    ...
  }
}
```

(i.e., manifest `d54b` refers to manifest `5b0b`)

...then the response header should be `OCI-Subject: sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270`

This means the registry had to at least parse that JSON and extract the `subject` to be able to return it to you.

I like this.

What should the client do if the `OCI-Subject` response is present but not the subject it sent (`OCI-Subject: sha256:0000...`) -- fail probably? I don't think we need to add spec language for this, just curious mostly.

Reply:
I think probably fail or use the fallback tag.